star-navigation/src/components/simulation/simulation-view.tsx

```diff
@@ -1,18 +1,41 @@
 import React, { useEffect, useCallback, useRef, useState } from "react";
 import * as THREE from "three";
 import { clsx } from "clsx";
+import { getTimes } from "suncalc";
 import { config } from "../../config";
 import { IModelInputState, SNAPSHOT_REQUESTED } from "../../types";
 import { StarView } from "../star-view/star-view";
-import { daytimeOpacity } from "../../utils/daytime";
-import { getHeadingFromAssumedNorthStar, getStarHeadingFromNorth, invertCelestialSphereRotation } from "../../utils/sim-utils";
+import { dateToFractionalHoursInRightTimezone, getDateTimeString, getHeadingFromAssumedNorthStar, getStarHeadingFromNorth, invertCelestialSphereRotation } from "../../utils/sim-utils";
 import { Landscape } from "./landscape";
 import { Button } from "../button";
 
 import BackIcon from "../../assets/back-icon.svg";
 
 import css from "./simulation-view.scss";
 
+export const withinHour = (timeOfDay: number, fractionalHour: number) => {
+  return (timeOfDay >= Math.floor(fractionalHour)) && (timeOfDay < Math.ceil(fractionalHour));
+};
+
+export const daytimeOpacity = (inputState: Pick<IModelInputState, "month" | "timeOfDay" | "day">) => {
+  const date = new Date(getDateTimeString(inputState));
+  const times = getTimes(date, config.observerLat, config.observerLong);
+  const sunrise = dateToFractionalHoursInRightTimezone(times.sunrise);
+  const sunset = dateToFractionalHoursInRightTimezone(times.sunset);
+  const timeOfDay = inputState.timeOfDay;
+
+  let result: number;
+  if (withinHour(timeOfDay, sunrise) || withinHour(timeOfDay, sunset)) {
```

Comment:
This seems a bit crude if I'm reading it right. If sunrise is 5:59, then you get half-opacity from 5-6, whereas if it's 6:01 then it's from 6-7. Seems like it wouldn't be that hard to compute thirty minutes before and after sunrise/sunset, and the `suncalc` library even provides `goldenHourEnd` and `goldenHour` properties that might be useful.

Reply:
Yes, I agree. It's not my code, it's some leftovers coming from the FOSS repository.
When I add the twilight calculations to the new circular slider, this will be removed and replaced with some better math. I didn't do it now, as I need to look at available props and decide which ones we're gonna show on the new slider, etc., and sync that with this view.